Overhaul the dropdown component

davidknezic opened the pull request:
The dropdown component gets a major overhaul, where markup, styles and the plugin changes!

![dropdown](https://cloud.githubusercontent.com/assets/198988/11020015/f9458aee-8611-11e5-9cbe-9a701d9ca3f3.gif)

# Markup

In the markup there are new `data-*` attributes required to make it easier for us to follow the BEM principle of styling:

```jade
.dropdown(data-dropdown)
  .dropdown__label(data-dropdown-label)
    +svg('arrow-bottom', 'dropdown__label__icon')
    .dropdown__label__text(data-dropdown-text)
  select.dropdown__select(data-dropdown-select)
    option Schweiz
    option Regenbogenland
```

# Styles

The block is fully BEM compliant. There's no `--js` modifier anymore but rather an `is-enhanced` state.

In order to get the condensed dropdown component, it's necessary to tell the label element to be smaller:

```jade
.dropdown(data-dropdown)
  .dropdown__label.dropdown__label--condensed
    //...
```

# Plugin

The plugin is es6 now, it doesn't get or set any BEM classes except for the state class `is-enhanced`. It finds its elements with the data attributes `data-dropdown`, `data-dropdown-label`, `data-dropdown-text` and `data-dropdown-select`.

sventschui commented:
It is safe to use data-label in this use case but we should be consistent and talk about naming conventions for data- attributes.

We should more likely use sth. like `data-dropdown-label`.

Another problem we may face is the nesting of these data attributes. (`data-dropdown` -> `data-select` -> `data-dropdown` -> `data-select`) - Not really possible in this use case but yeah

davidknezic commented:
True, these are problems that you wouldn't encounter when you rebuild the interactions in React or Angular.
But I changed it anyway and updated the pull request description. Let's try to update the other components gradually.

Will merge now...
